src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -666,18 +666,30 @@ private async Task<bool> EnumerateFilesFromArtifactsProvider(TContext globalCont
                     continue;
                 }
 
-                if (artifact.SizeInBytes == 0)
+                long artifactSize = 0;
+                try
+                {
+                    artifactSize = artifact.SizeInBytes.Value;
+                }
+                catch (Exception e) when (e is IOException || e is ArgumentException)
```

Comment:
Why `ArgumentException`? I hope there's not a property getter (all that's in the try block) that throws that.

Also, is the API in the try block owned by us? If it does IO and can therefore fail, it shouldn't have been a property.